airflow/gcp/hooks/dataproc.py

```diff
@@ -366,126 +211,623 @@ def build(self) -> Dict:
+            specified as ``config.worker_config.num_instances``, and the ``PATCH`` request body would specify
+            the new value, as follows:
+
+            ::
+
+                 { "config":{ "workerConfig":{ "numInstances":"5" } } }
+
+            Similarly, to change the number of preemptible workers in a cluster to 5, the ``update_mask``
+            parameter would be ``config.secondary_worker_config.num_instances``, and the ``PATCH`` request
+            body would be set as follows:
+
+            ::
+
+                 { "config":{ "secondaryWorkerConfig":{ "numInstances":"5" } } }
+
+            If a dict is provided, it must be of the same form as the protobuf message
+            :class:`~google.cloud.dataproc_v1.types.FieldMask`
+        :type update_mask: Union[Dict, google.cloud.dataproc_v1.types.FieldMask]
+        :param graceful_decommission_timeout: Optional. Timeout for graceful YARN decomissioning. Graceful
+            decommissioning allows removing nodes from the cluster without interrupting jobs in progress.
+            Timeout specifies how long to wait for jobs in progress to finish before forcefully removing nodes
+            (and potentially interrupting jobs). Default timeout is 0 (for forceful decommission), and the
+            maximum allowed timeout is 1 day.
+
+            Only supported on Dataproc image versions 1.2 and higher.
+
+            If a dict is provided, it must be of the same form as the protobuf message
+            :class:`~google.cloud.dataproc_v1.types.Duration`
+        :type graceful_decommission_timeout: Union[Dict, google.cloud.dataproc_v1.types.Duration]
+        :param request_id: Optional. A unique id used to identify the request. If the server receives two
+            ``UpdateClusterRequest`` requests with the same id, then the second request will be ignored and
+            the first ``google.longrunning.Operation`` created and stored in the backend is returned.
+        :type request_id: str
+        :param retry: A retry object used to retry requests. If ``None`` is specified, requests will not be
+            retried.
+        :type retry: google.api_core.retry.Retry
+        :param timeout: The amount of time, in seconds, to wait for the request to complete. Note that if
+            ``retry`` is specified, the timeout applies to each individual attempt.
+        :type timeout: float
+        :param metadata: Additional metadata that is provided to the method.
+        :type metadata: Sequence[Tuple[str, str]]
+        """
+        client = self.get_cluster_client(location=region)
+        result = client.update_cluster(
+            project_id=project_id,
+            region=region,
+            cluster_name=cluster_name,
+            cluster=cluster,
+            update_mask=update_mask,
+            graceful_decommission_timeout=graceful_decommission_timeout,
```

Comment:
Is graceful_decomission_timeout still used or deprecated ?

Reply:
Check, it's still in use, my mistake.